In [3]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [9]:
total_acc_2022_by_vehicle = pd.read_csv('C:\\Users\\ksree\\Documents\\dda9\\PYTHON\\PROJECTS\\Kristen_Capstone\\data\\2022_all_accidents_by_mc.csv')

In [10]:
total_acc_2022_by_vehicle.head()

Vehicle Alabama  Alaska Arizona  \
0  Bus (Seats for 9-15 People, including driver)      26       0      32   
1  Bus (Seats for > 15 People, including driver)     122       1     121   
2              Single Unit Truck, 2 axle, 6 tire   1,024       4     220   
3                      Single Unit Truck, 3+axle     416       1     197   
4               Single Unit Truck, axles unknown       2       0     198   

  Arkansas California Colorado  Connecticut  Delaware  Dist. Of Columbia  ...  \
0       13         42        8            6         5                 25  ...   
1       80        740      139          219        65                 86  ...   
2      574      4,056      372          433       163                 60  ...   
3      185        819      248          187        90                 13  ...   
4        2         12        0           75         2                  2  ...   

  Tennessee  Texas  Utah  Vermont Virginia Washington West Virginia Wisconsin  \
0        18    211     1        0       73          7            13        22   
1       194    868    26        6      324        191            70       208   
2       858  3,356    90       27    1,146        396            67       418   
3     1,710  1,224    65       19      582        281            83       409   
4        39    172   234        7        9         27            14         0   

  Wyoming Sub Total  
0       2     2,055  
1      11    11,592  
2      62    35,651  
3      48    19,433  
4       2     3,674  

[5 rows x 53 columns]

In [11]:
total_acc_2022_by_vehicle[total_acc_2022_by_vehicle.columns[2:]]=total_acc_2022_by_vehicle[total_acc_2022_by_vehicle.columns[2:]].replace(',','',regex=True).astype('int64')

In [13]:
vehicle_highest_2022 = total_acc_2022_by_vehicle.nlargest(3, "Sub Total")

In [14]:
vehicle_highest_2022

Vehicle Alabama  Alaska  Arizona  Arkansas  \
7               Tractor/Semi-trailer   2,437       4     1257      2040   
2  Single Unit Truck, 2 axle, 6 tire   1,024       4      220       574   
3          Single Unit Truck, 3+axle     416       1      197       185   

   California  Colorado  Connecticut  Delaware  Dist. Of Columbia  ...  \
7        6956      1163          736       272                  0  ...   
2        4056       372          433       163                 60  ...   
3         819       248          187        90                 13  ...   

   Tennessee  Texas  Utah  Vermont  Virginia  Washington  West Virginia  \
7       1888  10003   428       47      2780         911            662   
2        858   3356    90       27      1146         396             67   
3       1710   1224    65       19       582         281             83   

   Wisconsin  Wyoming  Sub Total  
7       1343      862      83177  
2        418       62      35651  
3        409       48      19433  

[3 rows x 53 columns]

Tractor/Semi-trailer has the highest accidents in 2022 at 83k

In [15]:
vehicle_lowest_2022 = total_acc_2022_by_vehicle.nsmallest(3, "Sub Total")

In [16]:
vehicle_lowest_2022

Vehicle Alabama  Alaska  Arizona  \
10  Passenger Car, Only Displays HazMat Placard       0       0        0   
11     Light Truck, Only Display HazMat Placard       0       0        4   
9                                Tractor/Triple       0       0        4   

    Arkansas  California  Colorado  Connecticut  Delaware  Dist. Of Columbia  \
10         0           0         0            0         0                  0   
11         0           0         0            0         0                  0   
9          0           4         1            0         0                  0   

    ...  Tennessee  Texas  Utah  Vermont  Virginia  Washington  West Virginia  \
10  ...          0      0     0        0         0           0              0   
11  ...          0      0     0        0         1           0              2   
9   ...          3      8     4        0         0           0              0   

    Wisconsin  Wyoming  Sub Total  
10          0        0          1  
11          0        0         19  
9           1        0         89  

[3 rows x 53 columns]

Passenger Car has the lowest at 1, followed by light truck at 19, and tractor/triple at 89 for 2022

2021 accidents by MV

In [18]:
total_acc_2021_by_vehicle = pd.read_csv('C:\\Users\\ksree\\Documents\\dda9\\PYTHON\\PROJECTS\\Kristen_Capstone\\data\\2021_all_accidents_by_mc2.csv')

In [21]:
total_acc_2021_by_vehicle.head()

Vehicle Alabama  Alaska  Arizona  \
0  Bus (Seats for 9-15 People, including driver)      37       0       27   
1  Bus (Seats for > 15 People, including driver)      94       0      112   
2              Single Unit Truck, 2 axle, 6 tire   1,050       3      199   
3                      Single Unit Truck, 3+axle     517       5      186   
4               Single Unit Truck, axles unknown      28       0      191   

   Arkansas  California  Colorado  Connecticut  Delaware  Dist. Of Columbia  \
0        17          34         9            8        11                 20   
1        84         590       115          214        86                164   
2       611        4376       380          491       174                 96   
3       233         812       255          199       125                 26   
4         5         101         1            7         0                  0   

   ...  Tennessee  Texas  Utah  Vermont  Virginia  Washington  West Virginia  \
0  ...         23    167     3        3        56          14             12   
1  ...        198    893    24        6       259         166             59   
2  ...        901   4063    92       24       909         372             74   
3  ...       1785   1463    85       22       558         241             89   
4  ...         29    259   256        4        10          15             12   

   Wisconsin  Wyoming  Subtotal  
0         19        0      2105  
1        195       12     11412  
2        425       75     38883  
3        466       52     21054  
4          3        2      3984  

[5 rows x 53 columns]

In [20]:
total_acc_2021_by_vehicle[total_acc_2021_by_vehicle.columns[2:]]=total_acc_2021_by_vehicle[total_acc_2021_by_vehicle.columns[2:]].replace(',','',regex=True).astype('int64')

In [24]:
vehicle_highest_2021 = total_acc_2021_by_vehicle.nlargest(3, "Subtotal")

In [25]:
vehicle_highest_2021

Vehicle Alabama  Alaska  Arizona  Arkansas  \
7               Tractor/Semi-trailer   2,609       2     1467      2168   
2  Single Unit Truck, 2 axle, 6 tire   1,050       3      199       611   
3          Single Unit Truck, 3+axle     517       5      186       233   

   California  Colorado  Connecticut  Delaware  Dist. Of Columbia  ...  \
7        7560      1246          902       277                  0  ...   
2        4376       380          491       174                 96  ...   
3         812       255          199       125                 26  ...   

   Tennessee  Texas  Utah  Vermont  Virginia  Washington  West Virginia  \
7       2076  11709   396       35      2693        1128            728   
2        901   4063    92       24       909         372             74   
3       1785   1463    85       22       558         241             89   

   Wisconsin  Wyoming  Subtotal  
7       1571      902     93144  
2        425       75     38883  
3        466       52     21054  

[3 rows x 53 columns]

Tractor/Semi-trailer has the highest accidents in 2022 at 93k

In [26]:
vehicle_lowest_2021 = total_acc_2021_by_vehicle.nsmallest(3, "Subtotal")

In [27]:
vehicle_lowest_2021

Vehicle Alabama  Alaska  Arizona  \
10  Passenger Car, Only Displays HazMat Placard       0       0        0   
11     Light Truck, Only Display HazMat Placard       0       0        1   
9                                Tractor/Triple       0       0        1   

    Arkansas  California  Colorado  Connecticut  Delaware  Dist. Of Columbia  \
10         0           0         0            0         0                  0   
11         0           0         0            0         0                  0   
9          0           6         1            0         0                  0   

    ...  Tennessee  Texas  Utah  Vermont  Virginia  Washington  West Virginia  \
10  ...          0      0     0        0         0           0              0   
11  ...          0      0     0        0         0           0              0   
9   ...          1     17     5        1         0           0              0   

    Wisconsin  Wyoming  Subtotal  
10          0        0         0  
11          0        0        31  
9           0        0       104  

[3 rows x 53 columns]

Passenger with 0, light truck with 31, and tractor/triple at 104

2020 Motor Vehicle accidents

In [28]:
total_acc_2020_by_vehicle = pd.read_csv('C:\\Users\\ksree\\Documents\\dda9\\PYTHON\\PROJECTS\\Kristen_Capstone\\data\\2020_all_accidents_by_mc1.csv')

In [29]:
total_acc_2020_by_vehicle

Vehicle Alabama  Alaska Arizona  \
0   Bus (Seats for 9-15 People, including driver)      16       0      18   
1   Bus (Seats for > 15 People, including driver)      92       0      88   
2               Single Unit Truck, 2 axle, 6 tire   1,000       2     192   
3                       Single Unit Truck, 3+axle     441       1     295   
4                Single Unit Truck, axles unknown      16       2     168   
5                                  Truck/Trailers     274       1     437   
6                         Truck/Tractor (bobtail)      99       0      31   
7                            Tractor/Semi-trailer   2,447       5   1,006   
8                                  Tractor/Double      34       1      71   
9                                  Tractor/Triple       0       0       2   
10    Passenger Car, Only Displays HazMat Placard       0       0       0   
11       Light Truck, Only Display HazMat Placard       0       0       1   
12                                        Missing       0       0       1   

   Arkansas California Colorado  Connecticut  Delaware  Dist. Of Columbia  \
0        20         46        2            5         9                  5   
1        77        379      131          129        42                118   
2       466      3,598      390          319       178                 96   
3       214        764      283          148       104                 49   
4         2         75        3            2         5                  5   
5       119        918       51           20        20                  8   
6        67        427       39            1        17                 11   
7     1,853      5,643    1,112          672       231                  3   
8        34        746       34           15         1                  0   
9         0         11        1            0         0                  0   
10        0          0        0            0         0                  0   
11        0          1        0            0         0                  0   
12        0          5        0            0         0                  1   

    ... Tennessee   Texas  Utah  Vermont Virginia Washington West Virginia  \
0   ...        19     146     4        1       38          2             5   
1   ...       134     705    33        5      145        148            47   
2   ...       663   3,106   215       25      729        328            65   
3   ...       832   1,281   102       29      445        247            85   
4   ...       195     172   298        3        6         16            12   
5   ...       722   3,128    12       54      146        349           121   
6   ...        51     547   695        1      136         26            26   
7   ...     1,771  10,262    18       50    2,260        827           630   
8   ...        91     162     2        2       81        109            24   
9   ...         3       5     0        0        0          0             0   
10  ...         0       0     0        0        0          0             0   
11  ...         1       0     0        1        1          0             0   
12  ...         0      21     0       11        1          0             1   

   Wisconsin Wyoming Sub Total  
0         15       3     1,741  
1        140       9     8,509  
2        416      77    33,012  
3        404      54    18,120  
4          1      10     4,206  
5         55      41    15,895  
6         38      10     4,348  
7      1,316     837    78,297  
8         28      74     2,895  
9          0       1       149  
10         0       0         1  
11         1       0        28  
12         0       0       493  

[13 rows x 53 columns]

In [32]:
total_acc_2020_by_vehicle[total_acc_2020_by_vehicle.columns[2:]]=total_acc_2020_by_vehicle[total_acc_2020_by_vehicle.columns[2:]].replace(',','',regex=True).astype('int64')

In [33]:
vehicle_highest_2020 = total_acc_2020_by_vehicle.nlargest(3, "Sub Total")

In [34]:
vehicle_highest_2020

Vehicle Alabama  Alaska  Arizona  Arkansas  \
7               Tractor/Semi-trailer   2,447       5     1006      1853   
2  Single Unit Truck, 2 axle, 6 tire   1,000       2      192       466   
3          Single Unit Truck, 3+axle     441       1      295       214   

   California  Colorado  Connecticut  Delaware  Dist. Of Columbia  ...  \
7        5643      1112          672       231                  3  ...   
2        3598       390          319       178                 96  ...   
3         764       283          148       104                 49  ...   

   Tennessee  Texas  Utah  Vermont  Virginia  Washington  West Virginia  \
7       1771  10262    18       50      2260         827            630   
2        663   3106   215       25       729         328             65   
3        832   1281   102       29       445         247             85   

   Wisconsin  Wyoming  Sub Total  
7       1316      837      78297  
2        416       77      33012  
3        404       54      18120  

[3 rows x 53 columns]

Tractor/Semi at 78k in 2020

In [35]:
vehicle_lowest_2020 = total_acc_2020_by_vehicle.nsmallest(3, "Sub Total")

In [36]:
vehicle_lowest_2020

Vehicle Alabama  Alaska  Arizona  \
10  Passenger Car, Only Displays HazMat Placard       0       0        0   
11     Light Truck, Only Display HazMat Placard       0       0        1   
9                                Tractor/Triple       0       0        2   

    Arkansas  California  Colorado  Connecticut  Delaware  Dist. Of Columbia  \
10         0           0         0            0         0                  0   
11         0           1         0            0         0                  0   
9          0          11         1            0         0                  0   

    ...  Tennessee  Texas  Utah  Vermont  Virginia  Washington  West Virginia  \
10  ...          0      0     0        0         0           0              0   
11  ...          1      0     0        1         1           0              0   
9   ...          3      5     0        0         0           0              0   

    Wisconsin  Wyoming  Sub Total  
10          0        0          1  
11          1        0         28  
9           0        1        149  

[3 rows x 53 columns]

Passenger with 1, light truck with 28, tractor/triple with 149

converting to csv's

In [37]:
vehicle_lowest_2020.to_csv('../csv/vehicle_lowest_2020.csv', index = False)

In [38]:
vehicle_lowest_2021.to_csv('../csv/vehicle_lowest_2021.csv', index = False)

In [39]:
vehicle_lowest_2022.to_csv('../csv/vehicle_lowest_2022.csv', index = False)

In [40]:
vehicle_highest_2020.to_csv('../csv/vehicle_highest_2020.csv', index = False)

In [41]:
vehicle_highest_2021.to_csv('../csv/vehicle_highest_2021.csv', index = False)

In [42]:
vehicle_highest_2022.to_csv('../csv/vehicle_highest_2022.csv', index = False)